extensions/common/vault/vault-azure/src/main/java/org/eclipse/edc/vault/azure/AzureVaultExtension.java

```diff
@@ -42,11 +56,45 @@ public String name() {
 
     @Provider
     public Vault createVault(ServiceExtensionContext context) {
-        var name = context.getConfig().getString(VAULT_NAME);
+        var config = context.getConfig();
+        var override = config.getString(VAULT_URL_OVERRIDE);
+
+        if (override != null && !override.isEmpty()) {
+            return createCustomVault(config, new SecretClientBuilder());
+        } else {
+            return createDefaultVault(config);
+        }
+    }
+
+    @NotNull
+    private AzureVault createDefaultVault(Config config) {
+        var name = config.getString(VAULT_NAME);
+        var credentials = new DefaultAzureCredentialBuilder().build();
         var client = new SecretClientBuilder()
                 .vaultUrl("https://" + name + ".vault.azure.net")
-                .credential(new DefaultAzureCredentialBuilder().build())
+                .credential(credentials)
                 .buildClient();
+
         return new AzureVault(monitor, client);
     }
+
+    @NotNull
+    public Vault createCustomVault(Config config, SecretClientBuilder builder) {
+        var override = config.getString(VAULT_URL_OVERRIDE);
+        var useUnsafe = config.getBoolean(VAULT_URL_OVERRIDE_UNSAFE);
+        var credentials = new DefaultAzureCredentialBuilder().build();
+
+        try {
+            new URL(override);
```

Comment:
if you do it that way, then please consider calling `toUri()`
```suggestion
            new URL(override).toURI();
```
and catch a `URISyntaxException`. Explanation: https://www.baeldung.com/java-validate-url#validate-url-using-jdk

Reply:
Added + updated the test to make the previous version `new URL("...")` fail.